modules/round/src/main/SocketMap.scala

```diff
@@ -1,28 +1,42 @@
 package lila.round
 
 import scala.concurrent.duration._
+import scala.math.{ log10, sqrt }
 
 import lila.game.Game
-import lila.hub.actorApi.map.{ Tell, TellIfExists, Exists }
 import lila.hub.actorApi.Deploy
+import lila.hub.actorApi.map.{ Tell, TellIfExists, Exists }
+import lila.user.User
 
 private object SocketMap {
 
   def make(
     makeHistory: Game.ID => History,
     dependencies: RoundSocket.Dependencies,
-    socketTimeout: FiniteDuration
+    socketTimeout: FiniteDuration,
+    playban: lila.playban.PlaybanApi
   ): SocketMap = {
 
     import dependencies._
 
+    val defaultGoneWeight = fuccess(1f)
+    def goneWeight(userId: User.ID): Fu[Float] =
+      playban.sitAndDcCounter(userId) map { sc =>
+        if (sc > -5) 1f
+        else (1 - 0.7 * sqrt(log10(-sc - 3))).toFloat atLeast 0.25f
```

Comment:
Do we really want at least 25% though? That's 30 seconds in a Blitz game; for the real Bad Guys, I would be in favor of something like 5-10 seconds...

Reply:
We can go under 25% but not under 15s.

Reply:
Alright, 12.5% then? The lowest case there is 15s (for disconnection, not for ragequitting).